### PR TITLE
Implement SEPA Mandate Browser Switch

### PR DESCRIPTION
**Pull request by sarahkoop**
### Summary of changes

 - Implement browser switch to display SEPA Debit mandate 
 - Implement SEPADebitLifecycleObserver to encapsulate browser switching logic for the merchant
 - Parse success/error deep link URLs to prep for `tokenize` implementation (future PR)

 ### Checklist

 - ~[ ] Added a changelog entry~

- @jaxdesmarais 
- @sarahkoop

### CODE REVIEW

File: BraintreeCore/src/main/java/com/braintreepayments/api/BraintreeRequestCodes.java
Comment (sarahkoop) [inline BraintreeCore/src/main/java/com/braintreepayments/api/BraintreeRequestCodes.java:30]: We just incremented the previous request code, but is there any info on how these numbers were determined or what this new one should be?

File: Demo/src/main/java/com/braintreepayments/demo/DemoActivity.java
Comment (sarahkoop) [inline Demo/src/main/java/com/braintreepayments/demo/DemoActivity.java:66]: This is necessary until #523 is merged

File: Demo/src/main/java/com/braintreepayments/demo/SEPADebitFragment.java
Comment (sarahkoop) [inline Demo/src/main/java/com/braintreepayments/demo/SEPADebitFragment.java:53]: A specific IBAN/customer ID mandate can only be approved once, so randomly generating for continued testing.

File: SEPADebit/src/main/java/com/braintreepayments/api/SEPADebitRequest.java
Comment (sarahkoop) [inline SEPADebit/src/main/java/com/braintreepayments/api/SEPADebitRequest.java:46]: We wen't with the empty constructor/setter pattern to match most of our other request objects, but all params except merchant account ID are required. Should we add the required params to the constructor instead? Since they are required but technically could be null, should all have `@Nullable` instead of `@NonNull` annotation?

File: SEPADebit/src/main/java/com/braintreepayments/api/SEPADebitClient.java
Comment (sarahkoop) [inline SEPADebit/src/main/java/com/braintreepayments/api/SEPADebitClient.java:66]: This TODO was to align with the other client `setListener` methods in #523 - but this case is a little different, since we will call `tokenize` after browser switching. In this case a merchant will have to set the listener to receive results from tokenize - does that seem like the right approach?

File: SEPADebit/src/main/java/com/braintreepayments/api/SEPADebitRequest.java
Comment (sshropshire) [reply SEPADebit/src/main/java/com/braintreepayments/api/SEPADebitRequest.java:46]: I'm up for either, but honestly I do like the empty constructor pattern. Do we callback an exception when a param is missing?

Comment (sshropshire) [reply SEPADebit/src/main/java/com/braintreepayments/api/SEPADebitRequest.java:46]: I'd say whichever choice makes the API most consistent is the best path to take.

**Review by sshropshire — APPROVED** [review]
🌊

**Review by cgdibble — APPROVED** [review]

File: SEPADebit/src/main/java/com/braintreepayments/api/SEPADebitRequest.java
Comment (sarahkoop) [reply SEPADebit/src/main/java/com/braintreepayments/api/SEPADebitRequest.java:46]: Ok will update - I agree leave the empty constructor, I think all the params should be nullable since they _could_ be null, and we'll add checks in the tokenize method to return error if any aren't set 👍

Comment (sarahkoop) [reply SEPADebit/src/main/java/com/braintreepayments/api/SEPADebitRequest.java]: Oh wait forgot the API would handle this and return a descriptive error so we left off the null check

**Pull request merged**
